daemon/networkdriver/portallocator/portallocator.go

```diff
@@ -59,6 +68,31 @@ func NewErrPortAlreadyAllocated(ip string, port int) ErrPortAlreadyAllocated {
 	}
 }
 
+func init() {
+	const param = "/proc/sys/net/ipv4/ip_local_port_range"
+
+	line, err := ioutil.ReadFile(param)
+	if err != nil {
+		log.Errorf("Failed to read %s kernel parameter: %s", param, err.Error())
+		return
+	}
+	var start, end int
+	n, err := fmt.Fscanf(strings.NewReader(string(line)), "%d\t%d", &start, &end)
```

Comment:
I had comment about `*os.File` as reader to `fmt.Fscanf`. You should use `os.Open` to get it.